src/number_theory/divisors.lean

```diff
@@ -233,6 +301,54 @@ begin
   apply h.ne_one.symm,
 end
 
+lemma proper_divisors_eq_singleton_one_iff_prime :
+  n.proper_divisors = {1} ↔ n.prime :=
+begin
+  split,
+  { intro h,
+    have h1 := mem_singleton.2 rfl,
+    rw [← h, mem_proper_divisors] at h1,
+    refine ⟨h1.2, _⟩,
+    intros m hdvd,
+    rw [← mem_singleton, ← h, mem_proper_divisors],
+    cases lt_or_eq_of_le (nat.le_of_dvd (lt_trans (nat.succ_pos _) h1.2) hdvd),
+    { left,
+      exact ⟨hdvd, h_1⟩ },
+    { right,
+      exact h_1 } },
+  intro h,
+  ext x,
+  simp only [mem_proper_divisors, mem_singleton],
+  split,
+  { rintro ⟨hdvd, hlt⟩,
+    cases h.2 x hdvd,
+    { exact h_1},
+    { exfalso,
+      apply ne_of_lt hlt h_1 } },
+  { rintro rfl,
+    refine ⟨one_dvd n, h.1⟩ }
+end
+
+lemma sum_proper_divisors_eq_one_iff_prime :
+  ∑ x in n.proper_divisors, x = 1 ↔ n.prime :=
+begin
```

Comment:
It feels like you should be able to golf this proof. Probably there are some useful subfacts about `proper_divisors`.